tools/aie2xclbin/aie2xclbin.cpp

```diff
@@ -0,0 +1,162 @@
+//===- aiecc.cpp -----------------------------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Xilinx Inc.
+//
+//===---------------------------------------------------------------------===//
+
+#include "XCLBinGen.h"
+#include "aie/Dialect/AIE/Transforms/AIEPasses.h"
+#include "aie/Dialect/AIEX/Transforms/AIEXPasses.h"
+#include "aie/InitialAllDialect.h"
+#include "aie/Targets/AIETargets.h"
+#include "configure.h"
+#include "mlir/Dialect/Affine/Passes.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/IR/OwningOpRef.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Target/LLVMIR/Dialect/Builtin/BuiltinToLLVMIRTranslation.h"
+#include "mlir/Target/LLVMIR/Dialect/LLVMIR/LLVMToLLVMIRTranslation.h"
+#include "mlir/Target/LLVMIR/Export.h"
+#include "mlir/Tools/mlir-translate/Translation.h"
+#include "llvm/ADT/SmallString.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/JSON.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/Program.h"
+#include "llvm/Support/SourceMgr.h"
+#include <iostream>
+#include <regex>
+#include <stdlib.h>
+#include <string>
+
+using namespace llvm;
+using namespace mlir;
+namespace sfs = std::filesystem;
```

Comment:
so what's the verdict on `std::filesystem` vs `llvm/Support/Filesystem`?

Reply:
I need to remove all usage of std::filesystem.

Reply:
The problem is that on windows the std::filesystem uses a wide char type internally, which breaks interop with most string stuff. There was a discussion about this on discord and we decided to just avoid std::filesystem.

Reply:
That makes sense to use the LLVM solution then. But this should appear in some comment explaining why because it is not obvious to the reader.